### ISSUE TITLE
a performance issue after running the scans for multiple modules

### ISSUE THREAD

**Issue by iranimij** [yireo/Yireo_ExtensionChecker#45]
Currently after running the below command 

```
bin/magento mod:st --enabled | grep Yireo_ | while read MODULE ; do 
    echo "Checking $MODULE"
    bin/magento yireo_extensionchecker:scan --module $MODULE --hide-deprecated 1
done
```

we are running bin/magento command multiple times, in a complete installation of magento there is no issue with this, but when we do not have cache, running each bin/magento takes considerable time, so If we just run bin/magento one time and add all module names as parameters would be so faster

Something like this : 

```
 bin/magento yireo_extensionchecker:scan --module first_moduloe, second_mode, next_module
```

We have had this issue in our pipeline where we were using extension_checker for our modules in a magento without any cache. and it took a long time for it to be handled.

I have prepared this PR

https://github.com/yireo/Yireo_ExtensionChecker/pull/46

And we can use below command to cjecl all modules :

`bin/magento yireo_extensionchecker:scan --module $(bin/magento module:status --enabled | awk '{printf "%s%s",sep,$0; sep=","} END{print""}') --hide-needless 1 --hide-deprecated 1`

**Comment by iranimij** [comment]
Any updates?

**Comment by iranimij** [comment]
I also have a suggestion:

We can add a new option:
for example
`bin/magento yireo_extensionchecker:scan:modules --path 'path/to/the/directory/of/all/modules'` and then in this path `path/to/the/directory/of/all/modules` we can check and get list of all modules programatically and then scan all of them.

What's you idea?

**Comment by jissereitsma** [comment]
@iranimij Sorry for the delay, I've been wrapped in 1001 projects and needed to be picky on which one to invest my time.

Anyway, I'll look into the PR most likely next week. To already comment upon your latest suggestion: I'd rather have the argument `--module` to be used together with a comma-separated list of modules (like you suggested earlier) than abusing `--path` for this. Including a path with multiple modules would lead into further headaches like excluding modules from that inclusion, etc.

**Comment by jissereitsma** [comment]
I just merged the PR and will release a new version within the hour. Let me know if there were something else on your mind as well. Otherwise we can close this issue.

**Comment by iranimij** [comment]
@jissereitsma That's great that we can now check all modules by just running one `bin/magento` command, but we still need something to add to the module, currently if multiple modules have dependency issues we can not distinguish which error message is related to which module, so I have prepared a new PR so if we merge this PR we can see which error message is related to which module name.
Would you please review : 
https://github.com/yireo/Yireo_ExtensionChecker/pull/47

**Comment by sprankhub** [comment]
I think this issue can now be closed.

**Comment by iranimij** [comment]
@jissereitsma Would you please prepare a new release for these new changes? https://github.com/yireo/Yireo_ExtensionChecker/pull/47

**Comment by jissereitsma** [comment]
I released 2.3.1 now.

**Comment by iranimij** [comment]
@jissereitsma Would you please check my PR ?!
